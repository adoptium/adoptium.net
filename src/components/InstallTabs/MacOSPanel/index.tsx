import React from "react"
import ShellBox from "@/components/ShellBox"
import "../InstallTabs.css"

export const PureMacOSPanel = (): React.ReactNode => {
  // Hardcoded value for latest LTS, should be replaced with actual data fetching in Next.js
  const mostRecentLts = "21"
  return (
    <div>
      <ShellBox>
        <span className="install__text__no-select">
          # Install the latest LTS version
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">brew install --cask </span>
        temurin@{mostRecentLts}
      </ShellBox>
      <br />
      <ShellBox>
        <span className="install__text__no-select">
          # Install a specific version
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">brew install --cask </span>
        temurin@8
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">brew install --cask </span>
        temurin@{mostRecentLts}
      </ShellBox>
      <br />
      <br />
      <br />
      <ShellBox>
        <span className="install__text__no-select">
          # Uninstall a specific version
        </span>
        <br />
        <span className="install__text__no-select">$</span>
        <span className="install-text-command">brew uninstall --cask </span>
        temurin@{mostRecentLts}
      </ShellBox>
      <a className="install__docs-button" href="https://docs.brew.sh/Manpage">
        Read documentation
      </a>
    </div>
  )
}

const MacOSPanel = (): React.ReactNode => {
  return <PureMacOSPanel />
}

export default MacOSPanel
