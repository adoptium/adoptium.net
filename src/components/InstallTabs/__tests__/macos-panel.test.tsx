import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import { PureMacOSPanel as MacOSPanel } from "../MacOSPanel"

describe("Tests for MacOSPanel component", () => {
  it("renders correctly", () => {
    const { container } = render(<MacOSPanel />)
    expect(container).toMatchSnapshot()
  })
})
