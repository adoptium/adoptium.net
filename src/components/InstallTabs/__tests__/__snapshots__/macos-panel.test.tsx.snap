// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for MacOSPanel component > renders correctly 1`] = `
<div>
  <div>
    <pre
      class="shell-box no-highlight"
    >
      <div
        class="shell-box-top"
      />
      <code
        class="shell-box-code"
      >
        <span
          class="install__text__no-select"
        >
          # Install the latest LTS version
        </span>
        <br />
        <span
          class="install__text__no-select"
        >
          $
        </span>
        <span
          class="install-text-command"
        >
          brew install --cask 
        </span>
        temurin
      </code>
    </pre>
    <br />
    <pre
      class="shell-box no-highlight"
    >
      <div
        class="shell-box-top"
      />
      <code
        class="shell-box-code"
      >
        <span
          class="install__text__no-select"
        >
          # Install a specific version
        </span>
        <br />
        <span
          class="install__text__no-select"
        >
          $
        </span>
        <span
          class="install-text-command"
        >
          brew install --cask 
        </span>
        temurin@8
        <br />
        <span
          class="install__text__no-select"
        >
          $
        </span>
        <span
          class="install-text-command"
        >
          brew install --cask 
        </span>
        temurin@
        21
      </code>
    </pre>
    <br />
    <br />
    <br />
    <pre
      class="shell-box no-highlight"
    >
      <div
        class="shell-box-top"
      />
      <code
        class="shell-box-code"
      >
        <span
          class="install__text__no-select"
        >
          # Uninstall a specific version
        </span>
        <br />
        <span
          class="install__text__no-select"
        >
          $
        </span>
        <span
          class="install-text-command"
        >
          brew uninstall --cask 
        </span>
        temurin@
        21
      </code>
    </pre>
    <a
      class="install__docs-button"
      href="https://docs.brew.sh/Manpage"
    >
      Read documentation
    </a>
  </div>
</div>
`;
